RiotSwiftUI/Modules/Common/Util/RoundedBorderTextEditor.swift

```diff
@@ -73,18 +76,32 @@ struct RoundedBorderTextEditor: View {
                         .foregroundColor(theme.colors.tertiaryContent)
                         .allowsHitTesting(false)
                 }
-                ThemableTextEditor(text: $text, onEditingChanged: { edit in
-                    self.editing = edit
-                    onEditingChanged?(edit)
-                })
-                .modifier(ClearViewModifier(alignment: .top, text: $text))
-                // Found no good solution here. Hidding next button for the moment
-//                .modifier(NextViewModifier(alignment: .bottomTrailing, isEditing: $editing))
-                .padding(EdgeInsets(top: 2, leading: 6, bottom: 0, trailing: 0))
-                .onChange(of: text, perform: { newText in
-                    onTextChanged?(newText)
-                })
+                if isEnabled {
+                    ThemableTextEditor(text: $text, onEditingChanged: { edit in
+                        self.editing = edit
+                        onEditingChanged?(edit)
+                    })
+                    .modifier(ClearViewModifier(alignment: .top, text: $text))
+                    // Found no good solution here. Hidding next button for the moment
+    //                .modifier(NextViewModifier(alignment: .bottomTrailing, isEditing: $editing))
+                    .padding(EdgeInsets(top: 2, leading: 6, bottom: 0, trailing: 0))
+                    .onChange(of: text, perform: { newText in
+                        onTextChanged?(newText)
+                    })
+                } else {
+                    ThemableTextEditor(text: $text, onEditingChanged: { edit in
+                        self.editing = edit
+                        onEditingChanged?(edit)
+                    })
+                    .padding(EdgeInsets(top: 2, leading: 6, bottom: 0, trailing: 6))
+                    .onChange(of: text, perform: { newText in
+                        onTextChanged?(newText)
+                    })
+                    .opacity(0.5)
+                    .allowsHitTesting(false)
```

Comment:
Looking at `ClearViewModifier`, it should hide itself when there is no text right? If so, then the `if isEnabled` could be removed and instead use ternaries for these two like below.
```suggestion
                    .opacity(isEnabled ? 1 : 0.5)
                    .allowsHitTesting(isEnabled)
```

Reply:
Not sure to follow you on this one as a text input can have some text set and be also disabled. I'm not sure we can avoid `if isEnabled` in this case.